cmd/zed/internal/lakemanage/manage.go

```diff
@@ -78,3 +82,37 @@ func getBranches(ctx context.Context, conf Config, lk lakeapi.Interface, logger
 	}
 	return branches, nil
 }
+
+func getPools(ctx context.Context, conf Config, lk lakeapi.Interface) ([]*pools.Config, error) {
+	pls, err := lakeapi.GetPools(ctx, lk)
+	if err != nil {
+		return nil, err
+	}
+	if len(conf.Pools) == 0 {
+		return pls, nil
+	}
+	m := map[ksuid.KSUID]struct{}{}
+	var out []*pools.Config
+	for _, c := range conf.Pools {
+		p := selectPool(c, pls)
+		if p == nil {
+			return nil, fmt.Errorf("pool %q not found", c.Pool)
+		}
+		if _, ok := m[p.ID]; ok {
+			return nil, fmt.Errorf("duplicate pool in configuration: %q", c.Pool)
+		}
+		m[p.ID] = struct{}{}
+		out = append(out, p)
+	}
```

Comment:
It may not be worth over-thinking, but if you already have the duplicate detection working I could rationalize keeping it in as a courtesy for the user. For instance, I could imagine an environment with lots of similarly-named pools that vary in some small way, e.g., a trailing `_1`, `_2`, etc. Then I could see myself typing out `-pool first-ong_pool_name_1` and cutting & pasting that several times with intent to manually edit/change that trailing character rather than cutting and pasting each one individually. Then I could see myself making a mistake and missing the edit on one, which means a duplicate reference would slip through. Having the tooling flag that for me sounds nicer than it going through silently and leaving me wondering for a bit why the query perf for one of my pools didn't improve as I thought it would. But I respect that needs to be weighed against the cost of having extra code in there just for such an unlikely scenario.

Anyway, this could all be a little moot since we all agreed that most users are likely to reach for the version of the command that just hits all the pools at once so this enhancement will mostly be for us internal users. But since the "Nit" label made it sound like a close call, I figured I'd say something.

Reply:
I'm with @philrz here I like the duplicate detection.